src/commands/app/upgrade.tsx

```diff
@@ -0,0 +1,170 @@
+    ).current;
+
+    const currentAppVersion = await getAppVersionFromUuid(
+      this.apiClient,
+      currentAppUuid,
+      currentAppVersionUuid,
+    );
+
+    const upgradeCandidates =
+      await this.apiClient.app.listUpdateCandidatesForAppversion({
+        appId: currentAppUuid,
+        baseAppVersionId: currentAppVersionUuid,
+      });
+
+    if (this.flags["show-candidates"]) {
+      process.addInfo(
+        <Text>
+          Your {currentApp.name} in Version ${currentAppVersion.externalVersion}{" "}
+          can currently be upgraded to the following Versions:
+        </Text>,
+      );
+      for (let i = 0; i < upgradeCandidates.data.length; i++) {
+        process.addInfo(
+          <Text>{upgradeCandidates.data[i].externalVersion}</Text>,
+        );
+      }
+      return;
+    }
+
+    let targetAppVersion;
+
+    if (this.flags["target-version"] == "latest") {
+      targetAppVersion =
+        await getLatestAvailableTargetAppVersionForAppVersionUpgradeCandidates(
+          this.apiClient,
+          currentAppUuid,
+          currentAppVersionUuid,
+        );
+      if (!targetAppVersion) {
+        process.complete(
+          <Success>
+            Your App is already at the lastest available or possible version. ✅
+          </Success>,
+        );
+      }
+    } else {
+      targetAppVersion = await getAvailableTargetAppVersionFromExternalVersion(
+        this.apiClient,
+        currentAppUuid,
+        currentAppVersionUuid,
+        this.flags["target-version"],
+      );
+      if (!targetAppVersion) {
+        process.error(
+          "Given target version is not a valid upgrade candidate for the current version of your app. \n" +
+            "You can check valid upgrade candidates for your apps with the flag '--show-candidates'",
+        );
+        return;
+      }
+    }
+
+    if (!targetAppVersion) {
+      process.error("Target App version could not be determined properly.");
+      return;
+    }
+
+    if (!this.flags.force) {
+      const confirmed = await process.addConfirmation(
+        <Text>
+          Confirm Upgrading {currentApp.name}{" "}
+          {currentAppVersion.externalVersion} (description here) to Version{" "}
+          {targetAppVersion.externalVersion}
+        </Text>,
+      );
+      if (!confirmed) {
+        process.addInfo(<Text>Upgrade will not be triggered.</Text>);
+        process.complete(<></>);
+
+        ux.exit(1);
+      }
+    } else {
+      process.addInfo(
+        <Text>
+          Commencing upgrade of {currentApp.name}
+          {currentAppVersion.externalVersion} (description here) to Version
+          {targetAppVersion.externalVersion}.
+        </Text>,
+      );
+    }
+
+    this.apiClient.app.patchAppinstallation({
+      appInstallationId,
+      data: { appVersionId: targetAppVersion.id },
+    });
+
+    process.complete(
+      <Success>The App Upgrade has been started. Buckle up! 🚀</Success>,
+    );
```

Comment:
Maybe also add a `--wait` flag (like for the `install [appname]` commands), in case someone wants to use this command synchronously?

Reply:
it's on the todo list, just wanted to get basic functionalities out of the way first and then look into how the waiting at that point actually works before I just copy&paste functionalities.